crates/consensus/src/transaction/envelope.rs

```diff
@@ -51,27 +51,48 @@ pub enum OpTxEnvelope {
 ///
 /// Compared to Ethereum it can tell whether the transaction is a deposit.
 pub trait OpTransaction {
-    /// Returns true if the transaction is a deposit.
+    /// An associated deposit transaction variant.
+    type Deposit: DepositTransaction;
```

Comment:
hmm do we need this AT, I think it's safe to assume that for op the txdeposit type will always be the same, because this one is specced by opstack?

this wouldn't solve the from field access problem for #542

the trait DepositTransaction is probably not very useful